### PR TITLE
Add stackdriver stub param

### PR DESCRIPTION
**Pull request by bianpengyuan**
**What this PR does / why we need it**:
Move all stackdriver stub related param into a struct. Consolidate different telemetry testing endpoint to a `secure_endpoint`. Also add an insecure endpoint option, which is going to be used for istio e2e test.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by bianpengyuan** [comment]
@kyessenov Please take a look. This adds an insecure endpoint which will unblock adding e2e test in istio/istio proxy. Setting up tls and certs in a really deployment is hard.

File: extensions/stackdriver/stackdriver.cc [hunk 1]
```diff
@@ -135,6 +103,26 @@ std::string getCACertFile() {
   return ca_cert_file;
 }
 
+// Get secure stackdriver endpoint for e2e testing.
+std::string getSecureEndpoint() {
+  std::string monitoring_service;
```
Comment (kyessenov) [inline extensions/stackdriver/stackdriver.cc]: nit: rename to `secure_endpoint`

File: extensions/stackdriver/metric/registry.cc [hunk 2]
```diff
@@ -36,38 +35,46 @@ constexpr char kStackdriverStatsAddress[] = "monitoring.googleapis.com";
...
+                                 ? kSTSSubjectTokenPath
+                                 : stub_option.test_token_path;
     ::Extensions::Stackdriver::Common::setSTSCallCredentialOptions(
-        &sts_options, sts_port, token_path);
+        &sts_options, stub_option.sts_port, token_path);
     auto call_creds = grpc::experimental::StsCredentials(sts_options);
     auto channel = ::grpc::CreateChannel(
-        test_monitoring_endpoint.empty() ? kStackdriverStatsAddress
-                                         : test_monitoring_endpoint,
+        stub_option.secure_endpoint.empty() ? kStackdriverStatsAddress
```
Comment (kyessenov) [inline extensions/stackdriver/metric/registry.cc]: nit: this is inconsistent, should use `default_endpoint`

**Comment by kyessenov** [comment]
/hold
(feel free to remove whenever you get the comments)

**Pull request merged**